packages/nlp/test/nlp.test.js

```diff
@@ -932,6 +1081,53 @@ describe('NLP', () => {
         },
       ]);
     });
+    test('It should extract a between rule and return longest string', async () => {
+      const nlp = new Nlp({ forceNER: true });
+      nlp.addNerBetweenCondition('en', 'entity', 'from', 'to');
```

Comment:
it works but the problem I see with this is approach is that the result is not so useful. Maybe would be better something like that:

```javascript
[
       {
          start: 18,
          end: 23,
          accuracy: 1,
          sourceText: 'Madrid,
          entity: 'entity',
          type: 'trim',
          subtype: 'between',
          utteranceText: 'Madrid,
          len: 6,
        },
        {
          start: 57,
          end: 65,
          accuracy: 1,
          sourceText: 'Barcelona',
          entity: 'entity',
          type: 'trim',
          subtype: 'between',
          utteranceText: 'Barcelona',
          len: 9,
        },
       {
          start: 18,
          end: 65,
          accuracy: 1,
          sourceText: 'Madrid to Barcelona and then back from Barcelona',
          entity: 'entity',
          type: 'trim',
          subtype: 'between',
          utteranceText: 'Madrid to Barcelona and then back from Barcelona',
          len: 48,
        },
]
```
A result like this would give enough autonomy to developer to decide what wants to do with the results. Also having more than one entity of the same type & subtype would be an indicator that further processing need to be done to desambiguate. What do you think?

Reply:
not for this PR, just to gather feedback from a developer point of view

Reply:
I also agree that I do not like that "longest string", but when I remember correctly I did not changed it to "how it was before". I just added the test to make sure my changes do not break things.

But beside adding it as "another match" (because the devloper now has 3 matched and no idea what do do with them programatically) I would more introduce another Rule type "betweenStrict" or such and this should then react in such a way